File: Unity network UDP/Assets/scripts/network/packet/MessagePacket.cs
using System;
using UnityEngine;
using System.Collections.Generic;
using System.Linq;
using System.Text;

class MessagePacket : IPacket
{

    private String pseudo, message;

    public MessagePacket() { }

    public MessagePacket(String pseudo, String message)
    {
        this.message = message;
        this.pseudo = pseudo;
    }

    void IPacket.read(DataBuffer data)
    {
        this.pseudo = data.getString();
        this.message = data.getString();
    }

    void IPacket.write(DataBuffer data)
    {
        data.put(Register.getId(typeof(MessagePacket)));
        data.put(this.pseudo);
        data.put(this.message);
    }

    void IPacket.manage(byte[] data,Client client)
    {
        Debug.Log(this.pseudo + " : " + this.message);
    }

}

File: Unity Server Project/src/fr/technicalgames/network/packet/MessagePacket.java
package fr.technicalgames.network.packet;

import java.net.*;

import fr.technicalgames.client.*;
import fr.technicalgames.io.*;
import fr.technicalgames.network.*;
import fr.technicalgames.network.common.*;

public class MessagePacket implements IPacket{

	private String pseudo, message;

    public MessagePacket() { }

    public MessagePacket(String pseudo, String message){
        this.message = message;
        this.pseudo = pseudo;
    }

    public void read(DataBuffer data){
        this.pseudo = data.getString();
        this.message = data.getString();
    }

    public void write(DataBuffer data){
        data.put(this.pseudo);
        data.put(this.message);
    }

    public void manage(Server server,DatagramSocket socket,Client client,IPacket packet){
        Log.println(Log.INFO, pseudo + " : " + message);
    }

}
